tools/CheckDates.scala

```diff
@@ -0,0 +1,110 @@
+import sys.process._
+
+object CheckDates {
+  val usage = """
+
+    This tool compares a copyright date in files with the date of last change in git.
+    If the message of last change contains "Documents date fixed" it will be ignored and 
+    we will take next commit.
+
+    analyzeDates - just shows if anything is ok
+    fixWrongDates - actual fixing
+
+    From root folder:
+    Usage: scala tools/CheckDates.scala analyzeDates
+           scala tools/CheckDates.scala fixWrongDates
+
+  """
+
+  def readFilesList(): Array[String] = {
+    val myval =  ("find . -name *.scala ! -path ./tools/*  -type f") !!
+    val files_array = myval.split("\n").map(_.toString).distinct
+    
+    return files_array
+  }
+
+  def readFile(pathToFile: String): String = {
+    val source = scala.io.Source.fromFile(pathToFile)
+    val lines = try source.mkString finally source.close()
+
+    return lines
+  }
+
+  def githubDate(pathToFile: String): String = {
+    val buf_dates = (s"git log -3  --format=%cd $pathToFile" !!)
+    val buf_messages = (s"git log -3  $pathToFile" !!)
+    val messages = buf_messages.split("commit").distinct.drop(1)
+    val dates = buf_dates.split("\\n").distinct
+    var gits_info = Array[(String, String)]()
+    for (a <- 0 to dates.length - 1) {
+      gits_info = gits_info :+ (messages(a),dates(a))
+    }
+    val res = gits_info.filter(!_._1.contains("Documents date fixed"))
+
+    return res(0)._2
+  }
+
+  def ifFileWasInGit(pathToFile: String): Boolean = {
+    val res_dates = (s"git log -3 $pathToFile" !!)
+    if(res_dates == "") {
+      return false
+    } else {
+      return true
+    }
+  }
+
+  def analyzeDates(): Array[(String, String, String)] = {
+    val files_array = readFilesList()
+    var data = Array[(String, String, String)]()
+    for(file_path <- files_array) {
+      if(readFile(file_path).contains("(c) 2003-") && ifFileWasInGit(file_path)) {
```

Comment:
It will be ignored.

For the most files we have next situation.
```
alfe@rt:~/mkdev/bigdata/scala/scala/src/library/scala/collection/generic$ git log --format=%cd Clearable.scala
Fri May 3 11:36:37 2013 -0700
Fri Nov 2 18:12:57 2012 +0100
Wed May 2 18:23:14 2012 -0700
```

First date in git - 2012, inside the file - 2003.
For future files, I may fix this. But now I don't know, what is the point?